src/proxy/inbound.rs

```diff
@@ -525,6 +530,28 @@ impl Inbound {
     ) -> Result<(SocketAddr, Option<TunnelRequest>, Vec<Arc<Service>>), Error> {
         // We always target the local workload IP as the destination. But we need to determine the port to send to.
         let target_ip = conn.dst.ip();
+        
+        // When ztunnel is serving its own metrics over mTLS - the target is the ztunnel itself
+        if local_workload.name.starts_with("ztunnel") {
+            let metrics_port = match hbone_addr {
+                HboneAddress::SocketAddr(addr) if matches!(cfg.stats_addr, ConfigAddress::SocketAddr(s) if addr.port() == s.port())
+                    || matches!(cfg.stats_addr, ConfigAddress::Localhost(_, p) if addr.port() == p) => {
+                    match cfg.stats_addr {
+                        ConfigAddress::Localhost(_, port) => port,
```

Comment:
We should not support sending to localhost. If they set stats address to localhosty, that means they don't want it accessible externally. We should not bypass that. 

Also, it makes this much simpler since we can just remove this code